docs/DeveloperGuide.md

```diff
@@ -1,34 +1,234 @@
+    - If the list of similar locations is empty, `RouteCommand#executeCommand()` performs step 4 given below, 
+    else it calls the static method, `Ui#printPossibleLocsMessage()`, to print the list of similar locations. 
+4. Calls static `BusData#possibleBuses()` to get a list of buses with their routes from the starting location to 
+the destination.
+   - `BusData#possibleBuses()` calls `Bus#getPossibleRoute()` to check for a possible route for the given bus number.
+   - `BusData#possibleBuses()` repeats this call for all bus numbers.
+
+The following sequence diagram explains the above steps when the user enters `/route loc1 /to loc2`.
+
+![Overview](DG_Diagrams/RouteCommand.png)
+
+The following sequence diagrams explain the interactions omitted in the main diagram.
+
+![executing command](DG_Diagrams/RouteCommandInternal.png)
+
+![bus data](DG_Diagrams/BusData.png)
+
+
+#### 2.2.2. Favourite command description modifier (`/descfav` Feature)
+
+`/descfav <index> /to <newDescription>` command allows the user to change the current description of their favourite command
+at location **index** in the list to **newDescription**.
+
+>Note: **index** is the index of the item in the list when the first number is indexed as 1. As a result, 
+>we access this element in the ArrayList using **index** - 1.
+
+The class diagram in the figure below shows how different classes used for implementation of the `/descfav` command 
+are linked to each other. 
+
+![DescFav class diagram](DG_Diagrams/descFavClass.png)
+
+The `DescFavCommand#executeCommand()` method of DescFavCommand Class executes the command in the following steps:
+1. Calls `DescFavParser#parseInput()` to check if the command message input by the user is valid.
+    - Throws an exception if 
+        + the input is empty.
+        + the delimiter `/to` is missing.
+    - Calls `Parser#splitCommands()` to separate the **index** and **description**.
+    - Checks the validity of each parameter and throws an exception if 
+        + both or either of the parameters have only spaces.
+        + the given index is not a number.
+    - If both inputs are valid, it assigns the input values to index and description variables.
+2. Calls `DescFavParser#getIndex()` and `DescFavParser#getDescription()` to get the appropriate values input by the user.
+3. Calls static `FavList#changeDesc()` to change the description of favourite at **index**.
+    - Calls static `FavList#checkIndexAndDesc()` which throws an exception if **index** is out of bounds or if 
+    **description** is the same as the previous description of this favourite.
+    - Calls `Fav#changeDesc()` to update the old description to **description**.
+    
+The following sequence diagram explains the above steps when the user enters `/descfav 1 /to hello`.
+
+![Overview](DG_Diagrams/descFav.png)
+
+The following sequence diagram explains the interactions omitted in the main diagram.
+
+![executing command](DG_Diagrams/descFavInternal.png)
```

Comment:
There isn't much use of destroy in the sequence diagrams, if they are named then this isn't a problem
Just a bit vague for the reader